h/services/auth_token.py

```diff
@@ -30,18 +30,24 @@ def validate(self, token_str):
         """
 
         if token_str in self._validate_cache:
-            return self._validate_cache[token_str]
+            token = self._validate_cache[token_str]
+            if token is not None and token.is_valid():
+                return token
+            return None
 
-        token = self._fetch_token(token_str)
+        token = self._fetch_auth_token(token_str)
         self._validate_cache[token_str] = token
         if token is not None and token.is_valid():
             return token
         return None
 
-    def _fetch_token(self, token_str):
-        token_model = (self._session.query(models.Token)
-                       .filter_by(value=token_str)
-                       .one_or_none())
+    def fetch(self, token_str):
```

Comment:
Given this is a new public method we’re adding to the service, should it have a docstring too? Pretty much what’s in the commit message, from the look of it.